docs/TASK.md

```diff
@@ -48,87 +48,102 @@ I am a user who wants to define my custom task and bring my model to Disco, with
+If you run the server yourself, you can use the two methods above but the prefered way is to **directly provide the task to the server before startup**. You can do this with the NPM [disco-server](https://www.npmjs.com/package/@epfml/disco-server) package without altering the code or recompiling it.
 
 ```js
-export * as cifar10 from './cifar10'
-export * as lus_covid from './lus_covid'
-export * as mnist from './mnist'
-export * as my_new_task from './my_new_task' // <---- including our new custom task!
-export * as titanic from './titanic'
+import { Disco, tf } from '@epfml/disco-server'
+
+// Define your own task provider (task definition + model)
+const customTask: TaskProvider = {
+    getTask(): Task {
+      return {
+        // Your task definition
+      }
+    },
+  
+    async getModel(): Promise<tf.LayersModel> {
+      const model = tf.sequential()
+      // Configure your model architechture
+      return model
+    }
+  }
+
+async function runServer() {
+  const disco = new Disco()
+  // Add your own custom task
+  await disco.addTask(customTask)
+  // Start the server
+  disco.serve()
+}
+
+runServer()
 ```
 
-This ensure that the task is properly exposed and thus the server will also be able to see and serve it.
+For more information, read the [server documentation](../server/README.md).
 
-> Note that `discojs-core` must only contain platform-agnostic code that works both in the browser and on Node.js.
-> Thus, if your task requires reading some file from your local file system, you need to define the task in `discojs-node` only: `discojs/discojs-node/src/tasks/my_new_task.ts`
-> In a similar fashion, tasks expecting to read from browser memory (such as IndexedDB or local storage) shall be defined in `discojs-web` instead: `discojs/discojs-web/src/tasks/my_new_task.ts`
+For your custom model, the JSON model architecture is necessary, but the .bin weight file is optional : if you include the weights file, your model will be loaded with the passed weights. If a weights file is not specified, the weights for the model will be initialized randomly.
 
-### Model
+For more detail about how to define a `Task` and a `tf.LayersModel` for your own `TaskProvider`, continue reading.
 
-Start by creating a function in `my_new_task.ts` called `model` that returns the `tf.LayersModel`. In this example we programatically 
-define our model.
 
-After exporting the Task, you need to also export a function called `model` that returns the layers model. If you use a 
-pre-trained model, you can simply load and return said model in the function via `tf.loadLayersModel(modelPath)`.
+### Model
 
-> Note, it's important to add the `export` tag to the model as well as to the subsequent objects that we define.
+The interface let you load your model however you want, as long as you return a `tf.LayersModel` at the end. If you use a 
+pre-trained model, you can simply load and return said model in the function via `tf.loadLayersModel(modelPath)`.
```

Comment:
(same comment as above, is the default recommendation some URL, or file-system access on the phone, or server?)